libs/go/aealite/wallet/utils.go

```diff
@@ -0,0 +1,398 @@
+	return sigBTC, err
+}
+
+// VerifyLedgerSignature verify signature of message using public key for supported ledgers
+func VerifyLedgerSignature(
+	ledgerId string,
+	message []byte,
+	signature string,
+	pubkey string,
+) (bool, error) {
+	verifySignature, found := verifyLedgerSignatureTable[ledgerId]
+	if found {
+		return verifySignature(message, signature, pubkey)
+	}
+	return false, errors.New("Unsupported ledger")
+}
+
+// VerifyFetchAISignatureBTC verify the RFC6967 string-encoded signature of message using FetchAI public key
+func VerifyFetchAISignatureBTC(message []byte, signature string, pubkey string) (bool, error) {
+	// construct verifying key
+	verifyKey, err := BTCPubKeyFromFetchAIPublicKey(pubkey)
+	if err != nil {
+		return false, err
+	}
+
+	// construct signature
+	signatureBTC, err := ParseFetchAISignature(signature)
+	if err != nil {
+		return false, err
+	}
+
+	// verify signature
+	messageHash := sha256.New()
+	_, err = messageHash.Write([]byte(message))
+	if err != nil {
+		return false, err
+	}
+
+	return signatureBTC.Verify(messageHash.Sum(nil), verifyKey), nil
+}
+
+// VerifyFetchAISignatureLibp2p verify RFC6967 string-encoded signature of message using FetchAI public key
+func VerifyFetchAISignatureLibp2p(message []byte, signature string, pubkey string) (bool, error) {
+	// construct verifying key
+	verifyKey, err := PubKeyFromFetchAIPublicKey(pubkey)
+	if err != nil {
+		return false, err
+	}
+
+	// Convert signature into DER encoding
+	sigBytes, err := base64.StdEncoding.DecodeString(signature)
+	if err != nil {
+		return false, err
+	}
+	sigDER := ConvertStrEncodedSignatureToDER(sigBytes)
+
+	// verify signature
+	return verifyKey.Verify(message, sigDER)
+}
+
+func SignFetchAI(message []byte, privKey string) (string, error) {
+	signingKey, _, err := KeyPairFromFetchAIKey(privKey)
+	if err != nil {
+		return "", err
+	}
+	signature, err := signingKey.Sign(message)
+	if err != nil {
+		return "", err
+	}
+	strSignature, err := ConvertDEREncodedSignatureToStr(signature)
+	if err != nil {
+		return "", err
+	}
+	encodedSignature := base64.StdEncoding.EncodeToString(strSignature)
+	return encodedSignature, nil
+}
+
+func signHashETH(data []byte) []byte {
+	msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(data), data)
+	return ethCrypto.Keccak256([]byte(msg))
+}
+
+// RecoverAddressFromEthereumSignature verify the signature and returns the address of the signer
+// references:
+//  - https://github.com/ethereum/go-ethereum/blob/55599ee95d4151a2502465e0afc7c47bd1acba77/internal/ethapi/api.go#L452-L459
+//  - https://github.com/ethereum/go-ethereum/blob/55599ee95d4151a2502465e0afc7c47bd1acba77/internal/ethapi/api.go#L404
+func RecoverAddressFromEthereumSignature(message []byte, signature string) (string, error) {
+	// prepare signature
+	sigBytes, err := hexutil.Decode(signature)
+	if err != nil {
+		return "", err
+	}
+
+	if sigBytes[64] != 27 && sigBytes[64] != 28 {
```

Comment:
keep 27 and 28 in constants , i dont the use of these numbers , but if other numbers come into picture or need to change them